cogito/protocol_test.go

```diff
@@ -102,6 +110,36 @@ func TestSourceValidationFailure(t *testing.T) {
 			},
 			wantErr: "source: invalid sink(s): [closed coffee shop]",
 		},
+		{
+			name: "no protocol prefix in git source github_api_endpoint",
+			source: cogito.Source{
+				Owner:             "the-owner",
+				Repo:              "the-repo",
+				AccessToken:       "the-token",
+				GithubApiEndpoint: "github.coffee.com/api/v3",
+			},
+			wantErr: "source: github_api_endpoint 'github.coffee.com/api/v3' is an invalid api endpoint",
+		},
+		{
+			name: "invalid http protocol prefix in git source github_api_endpoint",
+			source: cogito.Source{
+				Owner:             "the-owner",
+				Repo:              "the-repo",
+				AccessToken:       "the-token",
+				GithubApiEndpoint: "https:github.coffee.com/api/v3",
+			},
+			wantErr: "source: github_api_endpoint 'https:github.coffee.com/api/v3' is an invalid api endpoint",
+		},
+		{
+			name: "invalid http protocol prefix in git source github_api_endpoint",
+			source: cogito.Source{
+				Owner:             "the-owner",
+				Repo:              "the-repo",
+				AccessToken:       "the-token",
+				GithubApiEndpoint: "john.smith.cim",
+			},
+			wantErr: "source: github_api_endpoint 'john.smith.cim' is an invalid api endpoint",
```

Comment:
First of all, I don't like much the wording "protocol prefix". There is no prefix to a protocol.
Second, I don't understand why you have the third test. It should fail for the same reason the first one fails: no "protocol prefix".

Now, let's try to come up with a better wording. You're expecting the `GithubApiEndpoint` to hold a URI for the `https` _scheme_. The https scheme expect an _authority_ that must be introduced by `//`. After that their might be a _path_ that will start at first `/` after the authority.

https://datatracker.ietf.org/doc/html/rfc3986.html#section-3

Reply:
Yes, you are correct both first and third test, return the same error.
```
parse "github.coffee.com/api/v3": invalid URI for request
parse "john.smith.cim": invalid URI for request
```
I will remove the third one and adapt the wording according to the naming convention in the test name. Thanks

Reply:
Done as suggested.